code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -167,6 +167,56 @@
 	used_acid_spray = FALSE
 	to_chat(src, "<span class='notice'>You have produced enough acid to spray again.</span>")
 
+/atom/proc/acid_spray_act(mob/living/carbon/Xenomorph/X)
+	return TRUE
+
+/obj/structure/acid_spray_act(mob/living/carbon/Xenomorph/X)
+	if(!is_type_in_typecache(src, GLOB.acid_spray_hit))
+		return TRUE // normal density flag
+	health -= rand(40,60) + (X.upgrade * SPRAY_STRUCTURE_UPGRADE_BONUS)
+	update_health(TRUE)
+	return TRUE // normal density flag
+
+/obj/structure/razorwire/acid_spray_act(mob/living/carbon/Xenomorph/X)
+	. = ..()
+	. = FALSE // not normal density flag
```

Comment:
Wouldn't returning false do the same here?

Reply:
This is a fair statement for future generations to see that any inflexible rule will create stupid situations. Subversive.
:+1:

Reply:
Yes this is tongue in cheek coding